### DISCUSSION TITLE
Why the Connection ID and Stream ID are different between Android and windows?

### DISCUSSION THREAD

**Discussion by Ang0426** [microsoft/msquic#3491] [Q&A] [answered]
I'm  trying to send data between Android device and Windows device, and I find that when the msquic after connectd, start a Stream to send data, the Connection ID  and the Stream ID are not the same

on the Android device, the Connection ID is 0xb4000077081ad200 ,the Stream ID is 0xb4000076c22b8b50 ， and on the Windows device ,the Connection ID is 000001D6951AAF50 , the Stream ID is 000001D695E566B0.

I want know that what is the relationship with these IDs,  the stream is shut down soonly after  connection connected, Is it related to this？

And is it support Android send data with Window (I used  libmsquic.so on Android, and same version msquic.dll on Windows)?  Now it can connected but shut down soonly，thanks。

**Comment by anrossi** [comment, marked as answer]
Hello,
the IDs shared above look like memory addresses, not connection IDs, and not stream IDs.  A connection ID is a 20-byte identifier MsQuic generates automatically and isn't accessible via the API.  The stream ID is accessible via the API, with `GetParam` and the `QUIC_PARAM_STREAM_ID` parameter.

If the connection is shutdown shortly after connecting, make sure you set `PeerBidiStreamCount` or `PeerUnidiStreamCount` in `QUIC_SETTINGS` to a non-zero value, on the peer that is expected to receive streams. MsQuic only opens streams when a peer has explicitly allowed them. By default, no streams are allowed so the program can decide how many streams to allow.

**Reply by Ang0426** [reply]
Hi，
OK, I see, I will to check the GetParam API ,
thanks a lot for your help ,sir.
